vyxal/helpers.py

```diff
@@ -589,7 +589,9 @@ def gen():
 def primitive_type(item: Any) -> Union[str, type]:
     """Turns int/Rational/str into 'Scalar' and everything else
     into list"""
-    if type(item) in [int, sympy.Rational, str] or is_sympy(item):
+    if type(item) in [int, sympy.Rational, str, types.FunctionType] or is_sympy(
```

Comment:
You can also use `isinstance(item, (int, sympy.Rational, str, types.FunctionType))` for that `type ... in` check